website/src/common/VitessRefsCommand.tsx

```diff
@@ -0,0 +1,135 @@
+  const [selectedRefName, setSelectedRefName] = useState("");
+
+  const handleSelect = (vitessRef: VitessRefsData) => {
+    setInputValue(vitessRef.name);
+    setSelectedRefName(vitessRef.name);
+    setGitRef(vitessRef.commit_hash);
+    setOpen(false);
+  };
+
+  const handleInputKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      setGitRef(inputValue);
+      setSelectedRefName(inputValue);
+      setOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    const down = (e: KeyboardEvent) => {
+      if ((e.key === "k" && (e.metaKey || e.ctrlKey)) || e.key === "/") {
+        if (
+          (e.target instanceof HTMLElement && e.target.isContentEditable) ||
+          e.target instanceof HTMLInputElement ||
+          e.target instanceof HTMLTextAreaElement ||
+          e.target instanceof HTMLSelectElement
+        ) {
+          return;
+        }
+
+        e.preventDefault();
+        setOpen((open) => !open);
+      }
+    };
+
+    document.addEventListener("keydown", down);
+    return () => document.removeEventListener("keydown", down);
+  }, []);
+
+  useEffect(() => {
+    setInputValue(gitRef);
+  }, [gitRef]);
+
+  return (
+    <>
+      <Button
+        variant="outline"
+        className={cn(
+          "relative justify-start rounded-[0.5rem] bg-muted/50 text-sm font-normal text-muted-foreground shadow-none sm:pr-4 w-full md:w-40 lg:w-64"
+        )}
+        onClick={() => setOpen(true)}
+      >
+        <span className="hidden lg:inline-flex overflow-hidden">
+          {selectedRefName || gitRef || inputLabel}
+        </span>
+        <span className="inline-flex lg:hidden overflow-hidden">
+          {selectedRefName || gitRef || "Search..."}
+        </span>
+      </Button>
+      <CommandDialog open={open} onOpenChange={setOpen}>
+        <CommandInput
+          placeholder={inputLabel}
+          value={inputValue}
+          onInput={(e: ChangeEvent<HTMLInputElement>) =>
+            setInputValue(e.target.value)
+          }
+          onKeyDown={handleInputKeyDown}
+          className="w-full max-w-md mx-auto sm:max-w-lg lg:max-w-xl"
+        />
+        <CommandList>
+          <CommandEmpty>No results found.</CommandEmpty>
+          <CommandGroup heading="Branches">
+            {vitessRefs?.branches?.map((ref) => (
+              <CommandItem key={ref.name} onSelect={() => handleSelect(ref)}>
+                <span>{ref.name}</span>
+                <span className="hidden">{ref.commit_hash}</span>
+              </CommandItem>
+            ))}
+          </CommandGroup>
+          <CommandGroup heading="Releases">
+            {vitessRefs?.tags?.map((ref, index) => (
+              <CommandItem key={index} onSelect={() => handleSelect(ref)}>
+                <span>{ref.name}</span>
+                <span hidden>{ref.commit_hash}</span>
+              </CommandItem>
+            ))}
+          </CommandGroup>
+        </CommandList>
```

Comment:
Let's remove `Branches` and `Releases` in the modal when there are no result.

<img width="754" alt="image" src="https://github.com/user-attachments/assets/fff24a81-d32a-43da-aee5-1e4eb1adfd46">